operate/webapp/src/main/java/io/camunda/operate/webapp/security/permission/PermissionsService.java

```diff
@@ -0,0 +1,284 @@
+  }
+
+  /**
+   * getDecisionsWithPermission
+   *
+   * @return decisions for which the user has the given permission; the result matches either all
+   *     decisions, or a list of decisionId
+   */
+  public ResourcesAllowed getDecisionsWithPermission(final IdentityPermission identityPermission) {
+    return getResourcesWithPermission(
+        AuthorizationResourceType.DECISION_DEFINITION, identityPermission);
+  }
+
+  /**
+   * getResourcesWithPermission
+   *
+   * @return resources for which the user has the given permission; the result matches either all
+   *     resources, or a list of resourceIds
+   */
+  public ResourcesAllowed getResourcesWithPermission(
+      final AuthorizationResourceType resourceType, final IdentityPermission identityPermission) {
+    if (!permissionsEnabled()) {
+      return ResourcesAllowed.all();
+    }
+    if (!isAuthorized()) {
+      return ResourcesAllowed.withIds(Set.of());
+    }
+    if (identityPermission == null) {
+      throw new IllegalStateException("Identity permission cannot be null");
+    }
+    final PermissionType permissionType = getPermission(identityPermission);
+    final Authorization authorization = new Authorization(resourceType, permissionType);
+    final SecurityContext securityContext = getSecurityContext(authorization);
+    final List<String> ids = authorizationChecker.retrieveAuthorizedResourceKeys(securityContext);
+
+    return ResourcesAllowed.withIds(new LinkedHashSet<>(ids));
+  }
+
+  /**
+   * @return true if permissions checks are enabled
+   */
+  public boolean permissionsEnabled() {
+    return securityConfiguration.getAuthorizations().isEnabled();
+  }
+
+  private boolean isAuthorized() {
+    return (getAuthenticatedUserKey() != null);
+  }
+
+  private Long getAuthenticatedUserKey() {
+    final Authentication requestAuthentication =
+        SecurityContextHolder.getContext().getAuthentication();
+    if (requestAuthentication != null) {
+      final Object principal = requestAuthentication.getPrincipal();
+      if (principal instanceof final CamundaUser authenticatedPrincipal) {
+        return authenticatedPrincipal.getUserKey();
+      }
+    }
+    return null;
+  }
+
+  private boolean isAuthorizedFor(
+      String resourceId, AuthorizationResourceType resourceType, PermissionType permissionType) {
+    final Authorization authorization = new Authorization(resourceType, permissionType);
+    final SecurityContext securityContext = getSecurityContext(authorization);
+    return authorizationChecker.isAuthorized(resourceId, securityContext);
+  }
+
+  private SecurityContext getSecurityContext(Authorization authorization) {
+    return new SecurityContext(getAuthentication(), authorization);
+  }
+
+  private io.camunda.security.auth.Authentication getAuthentication() {
+    final Long authenticatedUserKey = getAuthenticatedUserKey();
+    final List<String> authorizedTenants = tenantsService.tenantIds();
+    // groups and roles will come later
```

Comment:
can you create a GH issue for this, so that we don't forget ?

Reply:
Added here: https://github.com/camunda/camunda/issues/25282